Lock versions for protobuf gen libraries.

danielmai opened the pull request:
* Clone golang/protobuf and gogo/protobuf libraries to latest released
  versions:
  * golang/protobuf: v1.3.2
  * gogo/protobuf: v1.2.1
* Regenerate protos/pb/pb.pb.go.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/dgraph/3752)
<!-- Reviewable:end -->

mangalaman93 commented on protos/Makefile:
```diff
@@ -26,16 +30,31 @@ help:
+
+.PHONY: golang_protobuf
+golang_protobuf:
+	@echo "\033[0;31mRemoving $(GOPATH)/src/github.com/golang/protobuf\033[0m"
```
The color codes are unnecessary and do not make much difference to the output. Can avoid it.

mangalaman93 commented on protos/Makefile:
```diff
@@ -26,16 +30,31 @@ help:
 .PHONY: clean
 clean:
 	@mkdir -p pb && rm -f pb/pb.pb.go
+	@rm -f $(GOBIN)/protoc-gen-go
```
`regenerate` is doing `check clean golang_protobuf gogo_protobuf` and `clean` is Phony. So every step will run every time which may be fine because we won't be running make regenerate too often. Could be problematic while I making changes to the proto file though.

danielmai commented:
___
*[protos/Makefile, line 42 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/3752#-LlY94-EmstykYIUQ5Xu-r1-42:-LlYZLImD8nx6KGDnWv_:b4sqs8m) ([raw file](https://github.com/dgraph-io/dgraph/blob/5ee54c95ab312255db1a772936c08ac8fc61aa0e/protos/Makefile#L42)):*
<details><summary><i>Previously, mangalaman93 (Aman Mangal) wrote…</i></summary><blockquote>

The color codes are unnecessary and do not make much difference to the output. Can avoid it.
</blockquote></details>

Done. Removed the colors.


<!-- Sent from Reviewable.io -->

danielmai commented:
___
*[protos/depcheck.sh, line 7 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/3752#-LlYCS5LDydmZclJYi6p:-LlYZOjm-wOn5u5o73e2:b2phns) ([raw file](https://github.com/dgraph-io/dgraph/blob/5ee54c95ab312255db1a772936c08ac8fc61aa0e/protos/depcheck.sh#L7)):*
<details><summary><i>Previously, martinmr (Martin Martinez Rivera) wrote…</i></summary><blockquote>

What's the difference between these two constants. The names are pretty similar so it's a bit confusing.
It looks like the versions of go and gogo but a comment would be helpful.
</blockquote></details>

Done. Renamed the variables and added comments. Should be clearer now.


<!-- Sent from Reviewable.io -->

danielmai commented:
___
*[protos/Makefile, line 33 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/3752#-LlY94-EmstykYIUQ5Xu-r1-33:-LlYZ_cgDYT8l-r8APTL:b-jhbbzq) ([raw file](https://github.com/dgraph-io/dgraph/blob/5ee54c95ab312255db1a772936c08ac8fc61aa0e/protos/Makefile#L33)):*
<details><summary><i>Previously, mangalaman93 (Aman Mangal) wrote…</i></summary><blockquote>

`regenerate` is doing `check clean golang_protobuf gogo_protobuf` and `clean` is Phony. So every step will run every time which may be fine because we won't be running make regenerate too often. Could be problematic while I making changes to the proto file though.
</blockquote></details>

Every directive in the Makefile is .PHONY. Is there an issue with removing the generated code and protoc-gen-* binaries when running `make regenerate`?


<!-- Sent from Reviewable.io -->

danielmai commented:
Protobuf-tooling version locking is done by #5291. It installs the versions specified in go.mod. Closing this PR.
